src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -28,6 +34,32 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
     };
   }
 
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+  async componentDidMount() {
```

Comment:
just curious: why do we need componentDidMount and componentWillUnmount for this change?

Reply:
We need to check the props.selectedOption.label exist, if not, need to call getDataSourceById to get the title back and set this value, since we only have one option, we only need to call at most once before rendering, this is the reason why put a componentDidMount here. As for `this._isMounted = false;` in `componentWillUnmount`, this refer to code here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source_management/public/components/data_source_selector/data_source_selector.tsx#L65